api/src/main/java/io/minio/PutObjectArgs.java

```diff
@@ -35,7 +37,13 @@ public BufferedInputStream stream() {
    */
   public String contentType() throws IOException {
     String contentType = super.contentType();
-    return (contentType != null) ? contentType : "application/octet-stream";
+    if (contentType != null) {
+      return contentType;
+    }
+    contentType = Files.probeContentType(Paths.get(super.object()));
```

Comment:
`putObject()` method is to upload `stream` to an object in S3 service. It doesn't matter whether a local file has object name or not. If your requirement is to upload a file to an object in S3 service, use `uploadObject()` method. This method detects content-type of the file.

Reply:
then it seems that application/octet-stream is the correct Content-Type when using putObject

Reply:
Yep. If content-type is not provided, it defaults to `application/octet-stream`.